pkg/storage/etcd/etcd_helper.go

```diff
@@ -558,6 +576,15 @@ func (h *etcdHelper) GuaranteedUpdate(ctx context.Context, key string, ptrToType
 			ttl = *newTTL
 		}
 
+		// Since update object may have a resourceVersion set, we need to clear it here.
+		if h.versioner != nil {
+			if err := h.versioner.UpdateObject(ret, meta.Expiration, 0); err != nil {
+				return errors.New("resourceVersion cannot be set on objects store in etcd")
+			}
+		} else {
+			return errors.New("couldn't access version without versioner")
```

Comment:
@smarterclayton - so IIUC, you suggest that "storing an object with resourceVersion in etcd if there is no versioner" (which means that we will have no-op updates), is better than "fail if there is no versioner" for now, right?